ouroboros-consensus/src/Ouroboros/Consensus/Node.hs

```diff
@@ -273,22 +273,26 @@ initNetwork
   -> IO ()
 initNetwork registry nodeArgs kernel RunNetworkArgs{..} = do
     -- serve local clients (including tx submission)
-    localServer <- forkLinked registry runLocalServer
+    localServer <- forkThread registry runLocalServer
```

Comment:
Would it sense to add `forkLinkedThread`?

Reply:
It seemed unnecessary, but since both you and @nfrisby prefer it, I'll add it :)

Reply:
I'll remove the `forkThreadSelf`, I thought we'd need it in tests but turns out it doesn't help. And it avoids requiring a `forkLinkedThreadSelf` variant then too.